lib/queue.js

```diff
@@ -884,6 +884,11 @@ Queue.prototype.retryJob = function(job) {
 };
 
 Queue.prototype.toKey = function(queueType){
+  return [this.keyPrefix ? '' : 'bull', this.name, queueType].join(':');
+};
+
+Queue.prototype.toPrefixedKey = function(queueType){
+  if (!this.keyPrefix) return this.toKey(queueType);
   return [this.keyPrefix, this.name, queueType].join(':');
 };
 
```

Comment:
So the bug is pretty subtle and you can replicate it by running the old code against the updated tests. You will see that some of the keys in redis have the prefix applied twice.

The reason is that the keyPrefix option in ioredis is a bit magical (transparent?) and you have to distinguish between calling ioredis command and constructing a key within a lua script or from a value stored in redis.

So to get around that, if no keyPrefix option is passed in these functions preserve the default behaviour and both return the same thing, but if keyPrefix exists, the rest of the code base has been updated to call the appropriate function and avoid the keyPrefix being applied twice.

Generally, if you are calling an ioredis function, you need to use the non-prefixed key, but if you are constructing a string to be used in a lua script, you need to use the prefixed key. For extra subtle complexity, the exception to the above is that ioredis also magically applies keyprefix to all the key args when calling 'evalsha', so in those instances you need to use the non-prefixed key.